src/etc/inc/filter.inc

```diff
@@ -446,6 +446,12 @@ function filter_configure_sync($verbose = false)
         $limitrules .= "set limit table-entries {$config['system']['maximumtableentries']}\n";
     }
 
+    if ($config['system']['rulesoptimization'] <> "") {
```

Comment:
@phpb-com better to check for empty($config['system']['rulesoptimization']) here, rulesoptimization is optional available and may not be initialised.